c7n/resources/dynamodb.py

```diff
@@ -106,20 +107,107 @@ class KmsFilter(KmsRelatedFilter):
+                    key: c7n:AliasName
+                    value: "^(alias/aws/dynamodb)"
+                    op: regex
     """
     RelatedIdsExpression = 'SSEDescription.KMSMasterKeyArn'
 
 
+@Table.filter_registry.register('continuous-backup')
+class TableContinuousBackupFilter(ValueFilter):
+    """Check for continuous backups and point in time recovery (PITR) on a dynamodb table.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: dynamodb-continuous-backups-disabled
+                resource: aws.dynamodb-table
+                filters:
+                  - type: continuous-backup
+                    key: ContinuousBackupsStatus
+                    op: eq
+                    value: DISABLED
+              - name: dynamodb-pitr-disabled
+                resource: aws.dynamodb-table
+                filters:
+                  - type: continuous-backup
+                    key: PointInTimeRecoveryDescription.PointInTimeRecoveryStatus
+                    op: ne
+                    value: ENABLED
+    """
+
+    annotation_key = 'c7n:continuous-backup'
+    annotate = False
+    schema = type_schema('continuous-backup', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('dynamodb:DescribeContinuousBackups',)
+
+    def process(self, resources, event=None):
+        self.augment([r for r in resources if self.annotation_key not in r])
+        return super(TableContinuousBackupFilter, self).process(resources, event)
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('dynamodb')
+        for r in resources:
+            try:
+                r[self.annotation_key] = client.describe_continuous_backups(
+                    TableName=r['TableName']).get('ContinuousBackupsDescription', {})
+            except client.exceptions.TableNotFoundException:
+                continue
+
+    def __call__(self, r):
+        return super(TableContinuousBackupFilter, self).__call__(r[self.annotation_key])
+
+
+@Table.action_registry.register('set-continuous-backup')
+class TableContinuousBackupAction(BaseAction):
+    """Set continuous backups and point in time recovery (PITR) on a dynamodb table.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: dynamodb-continuous-backups-disabled-set
+                resource: aws.dynamodb-table
+                filters:
+                  - type: continuous-backup
+                    key: ContinuousBackupsStatus
+                    op: eq
+                    value: DISABLED
+                actions:
+                  - type: set-continuous-backup
+
+    """
+
+    schema = type_schema(
+        'set-continuous-backup',
+        state={'type': 'boolean', 'default': True})
+    permissions = ('dynamodb:UpdateContinuousBackups',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('dynamodb')
```

Comment:
do we need a state filter here to ensure that the table is TableStatus: Active .. ie does this fail for deleting?  there's a method on actions.. filter_resources for this purpose incidentally.

Reply:
Good call, looks like it would fail if creating or deleting. I have added the recommended check in c20f5c4. Thanks!